### DISCUSSION TITLE
How to boostrap `poetry-plugin-tweak-dependencies-version`? 🤔

### DISCUSSION THREAD

**Discussion by kloczek** [sbrunner/poetry-plugin-tweak-dependencies-version#278] [General]
Looks like to build `poetry-plugin-tweak-dependencies-version` I must have already build and installed `poetry-plugin-tweak-dependencies-version` which additionally requires `poetry-plugin-drop-python-upper-constraint`
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting build dependencies for wheel...

ERROR Missing dependencies:
        poetry-plugin-tweak-dependencies-version
        poetry-plugin-drop-python-upper-constraint
```
and .. to build `poetry-plugin-drop-python-upper-constraint` I must have build and installed `poetry-plugin-tweak-dependencies-version`.
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting build dependencies for wheel...

ERROR Missing dependencies:
        poetry-plugin-tweak-dependencies-version
```
🤔

**Comment by sbrunner** [comment]
I use:
`pip install --use-pep517 --edit .`

**Reply by kloczek** [reply]
`builder` provides another pep517 based build procedure and as you see because how currently are defined build dependencies it is not possible to build `poetry-plugin-tweak-dependencies-version`.
Probably `pip` is not performing full pepp17 build procedure and it grabs just files to form .whl without checking build dependencies.
Do you see any solution in thi situation?

**Comment by kloczek** [comment]
I made the test with below patch:
```patch
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,5 +55,5 @@
 [tool.poetry-plugin-tweak-dependencies-version]
 default = "present"
 [build-system]
-requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning", "poetry-plugin-drop-python-upper-constraint", "poetry-plugin-tweak-dependencies-version"]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
 build-backend = "poetry.core.masonry.api"
```
And looks like both modules indeed are not needed and was able to build module without any issues.
```console
+ PBR_VERSION=1.5.2
+ PDM_BUILD_SCM_VERSION=1.5.2
+ SETUPTOOLS_SCM_PRETEND_VERSION=1.5.2
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting build dependencies for wheel...
* Building wheel...
Successfully built poetry_plugin_tweak_dependencies_version-0.1.0-py3-none-any.whl
```
Please let me know if you want this patch as PR.

**Comment by kloczek** [comment]
Ha .. version is wrong.
https://github.com/sbrunner/poetry-plugin-tweak-dependencies-version/blob/1b407e41f99a3a94b059cbf64fb6ceee02bc82aa/pyproject.toml#L10

**Reply by sbrunner** [reply]
No, because it's managed by poetry-dynamic-versioning...

**Comment by kloczek** [comment]
> No, because it's managed by poetry-dynamic-versioning...

Issue is that on top of what is possible to obtain from VCS is should be possible to inject version from some env variable when module is build from autogenerated from git tag tar ball which does not contain VCS metadata.
`setuptools-scm` uses for that `$SETUPTOOLS_SCM_PRETEND_VERSION` env variable, `pdm-backend` uses `$PDM_BUILD_SCM_VERSION`, and `pbr` uses `$PBR_VERSION`.
I cannot find in `poetry-dynamic-versioning` code something similar 🤔

**Comment by sbrunner** [comment]
`POETRY_DYNAMIC_VERSIONING_BYPASS` can be helpful :-)
